src/libs/actions/OnyxUpdateManager.ts

```diff
@@ -27,6 +28,132 @@ Onyx.connect({
+
+// This function will unpause the SequentialQueue, log an info and reset the deferred updates
+function unpauseQueueAndReset() {
+    console.debug('[OnyxUpdateManager] Done applying all updates');
+    deferredUpdates = {};
+    Onyx.set(ONYXKEYS.ONYX_UPDATES_FROM_SERVER, null);
+    SequentialQueue.unpause();
+}
+
+// This function applies a list of updates to Onyx in order and resolves when all updates have been applied
+const applyUpdates = (updates: DeferredUpdatesDictionary) => Promise.all(Object.values(updates).map((update) => OnyxUpdates.apply(update)));
+
+type DetectGapAndSplitResult = {applicableUpdates: DeferredUpdatesDictionary; updatesAfterGaps: DeferredUpdatesDictionary; latestMissingUpdateID: number | undefined};
+
+// In order for the deferred updates to be applied correctly in order,
+// we need to check if there are any gaps deferred updates.
+function detectGapsAndSplit(updates: DeferredUpdatesDictionary): DetectGapAndSplitResult {
+    const updateValues = Object.values(updates);
+    const applicableUpdates: DeferredUpdatesDictionary = {};
+
+    let gapExists = false;
+    let firstUpdateAfterGaps: number | undefined;
+
+    for (const [index, update] of updateValues.entries()) {
+        const isFirst = index === 0;
+
+        // If any update's previousUpdateID doesn't match the lastUpdateID from the previous update, the deferred updates aren't chained and there's a gap.
+        // For the first update, we need to check that the previousUpdateID of the fetched update is the same as the lastUpdateIDAppliedToClient.
+        // For any other updates, we need to check if the previousUpdateID of the current update is found in the deferred updates.
+        // If an update is chained, we can add it to the applicable updates.
+        const isChained = isFirst ? update.previousUpdateID === lastUpdateIDAppliedToClient : updates[Number(update.previousUpdateID)];
+        if (isChained) {
+            // If a gap exists already, we will not add any more updates to the applicable updates.
+            // Instead, we we have to find the first (chained) update after all gaps.
+            if (gapExists) {
+                // If "firstUpdateAfterGaps" hasn't been set yet and there was a gap, we need to set it to the first update after all gaps.
+                if (!firstUpdateAfterGaps) {
+                    firstUpdateAfterGaps = Number(update.previousUpdateID);
+                }
+            } else {
+                // If no gap exists yet, we can add the update to the applicable updates
+                applicableUpdates[Number(update.lastUpdateID)] = update;
+            }
+        } else {
+            // When we find a (new) gap, we need to set "gapExists" to true and reset the "firstUpdateAfterGaps" variable,
+            // so that we can continue searching for the next update after all gaps
+            gapExists = true;
+            firstUpdateAfterGaps = undefined;
```

Comment:
If we only have one update instead of N, we'll never fill the firstUpdateAfterGaps property, and will never add it to updatesAfterGaps, so we'll miss that update. I think we need to set the updateID here if isFirst is true.

Reply:
If that happens we'll never fill the `latestMissingUpdateID` property too.

Reply:
fixed!